CMakeLists.txt

```diff
@@ -1,182 +1,258 @@
 ####################################
-# General project definition
+# cmake Readout
 ####################################
 
-CMAKE_MINIMUM_REQUIRED(VERSION 3.5.2 FATAL_ERROR)
+# cmake settings
+CMAKE_MINIMUM_REQUIRED(VERSION 3.8.0 FATAL_ERROR)
+cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
 
-### CMP0025   Compiler id for Apple Clang is now AppleClang.
-### CMP0042   MACOSX_RPATH is enabled by default.
+# define project
+project(
+  Readout
+  DESCRIPTION "O2 Readout process"
+  LANGUAGES CXX
+)
 
-FOREACH (p
-        CMP0025 # CMake 3.0
-        CMP0042 # CMake 3.0
-        )
-    IF (POLICY ${p})
-        cmake_policy(SET ${p} NEW)
-    ENDIF ()
-endforeach ()
+# global compilation options
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake) # project specific cmake dir
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
+# define project directories
+# build-time binaries
+set(EXECUTABLE_OUTPUT_PATH "${CMAKE_BINARY_DIR}/bin")
+# local sources
+set(SOURCE_DIR src)
 
-project(Readout)
+# add local modules directory
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
-# Load some basic macros which are needed later on
-include(O2Utils)
-include(ReadoutDependencies)
+# add subdirectories
+add_subdirectory(doc)
 
-# Set the default build type to "RelWithDebInfo"
-if (NOT CMAKE_BUILD_TYPE)
-    set(CMAKE_BUILD_TYPE "RelWithDebInfo"
-            CACHE
-            STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel Coverage."
-            FORCE)
-endif (NOT CMAKE_BUILD_TYPE)
 
-# Set the version number of your project here (format is MAJOR.MINOR.PATCHLEVEL - e.g. 1.0.0)
-set(VERSION_MAJOR "0")
-set(VERSION_MINOR "0")
-set(VERSION_PATCH "0")
-set(VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH})
 
-# C++17
-set(CMAKE_CXX_STANDARD 17)
+###################################################
+# dependencies
+###################################################
 
+# dependencies to build Readout
+find_package(Common REQUIRED)
+find_package(InfoLogger REQUIRED)
+find_package(Configuration REQUIRED)
+find_package(Monitoring REQUIRED)
+find_package(ReadoutCard REQUIRED)
+find_package(Numa)
+find_package(FairMQ)
 
-# Add compiler flags for warnings and (more importantly) fPIC and debug symbols
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pedantic -Wextra -fPIC -g")
+# extract include directories from targets
```

Comment:
Hi @sy-c ,
Why do you need to extract include directories from targets? `target_link_libraries` should do the job.